crypto/stack/stack.c

```diff
@@ -191,7 +191,7 @@ static int internal_find(OPENSSL_STACK *st, const void *data,
                         ret_val_options);
     if (r == NULL)
         return (-1);
-    return (int)((char **)r - st->data);
+    return (int)((const char **)r - st->data);
```

Comment:
I'm suggesting that `r` should either be declared like this:
````
const char **r;
````
In which case no cast would be required in the return type (but possibly one in the `OBJ_bsearch_ex_` return instead)
Or it should be declared like this:
```
void *r;
````
The current declaration doesn't make a lot of sense when compared to how it is used.